src/webgl/p5.RendererGL.js

```diff
@@ -1781,6 +1776,15 @@ p5.RendererGL = class RendererGL extends Renderer {
     return point;
   }
 
+  _getImmediateImageShader() {
+    const imageShader = this.userImageShader;
+    if(imageShader) {
+      return imageShader;
+    }
+    return this._getImmediateModeShader() ; 
+  }
+  
+
```

Comment:
we could probably make just one method for all three, `_getImmediateImageShader()`, `_getImmediateFillShader()` and `_getRetainedFillShader()` and could name as `_getFillShader()` something like that. 

The only problem could be with the overriding of `imageShader()` with the beginShape/endShape part, so we could probably add a logic here: https://github.com/processing/p5.js/blob/581075435d6cb1e818f19102dc087e2068e75554/src/webgl/3d_primitives.js#L3437
something like:
```js
this._drawingImage = true;
this.beginShape();
// ...
this.endShape();
this._drawingImage = false;
```

and for the `_getFillShader()` I am talking about, it could look something like:
```js
getFillShader() {
  if (this._drawingImage) {
    return this.userImageShader;
  } else if (this.userFillShader) {
    return this.userFillShader;
  } else if ...
```

Really thanks to @davepagurek for his suggestions on this.:)